gremlin-dotnet/src/Gremlin.Net/Driver/Connection.cs

```diff
@@ -172,28 +170,35 @@ private string SaslArgument()
             return Convert.ToBase64String(authBytes);
         }
 
-        private void BeginSendingMessages()
+        private void Enqueue(RequestMessage message)
```

Comment:
To ensure that the counter increment and request queueing go always together in exactly one place in the code.

Reply:
Naming is always debatable though.

Reply:
> To ensure that the counter increment and request queueing go always together in exactly one place in the code.

Fair point, but then the name should reflect that the method not only enqueues the message, but also ensures that it will be sent. How about something like `EnsureMessageIsSent`? The name is not really elegant, but I think that it actually reflects what the method does.